resources/leiningen/new/brevity/src/brevity/clj/routes/middleware.clj

```diff
@@ -0,0 +1,16 @@
+(ns {{name}}.clj.routes.middleware
+  (:require [bidi.bidi :as bidi]))
+
+(defn wrap-bidi [handler wrapped-routes renderer]
+      (fn [{:keys [uri path-info] :as req}]
+          (let [path (or path-info uri)
+                resolved-route (bidi/match-route* wrapped-routes path req)
+                resolved-handler (:handler resolved-route)
+                route-params (:route-params resolved-route)]
+               (if (= :four-o-four resolved-handler)
```

Comment:
Should we just have it return 404 if there’s no `resolved-handler`?

Reply:
In that case it'll go down the chain - going to static resources and eventually the not-found handler.  I just made a change that lets the not-found handler still render the page with a 404, which will fall through to whatever component is mapped on the frontend for `:four-o-four`.

Reply:
What about just throwing an exception and clojure.lang.ExceptionInfo? I've been debating whether or not we should just have 4**'s throwing exceptions. Thoughts?

Reply:
In this case if a route isn't found, the middleware is relying on the fallthrough to `compojure.route/resources`.  I like the idea of completely eliminating 500's in production though - probably in the course of #6.